bpf/lib/icmp6.h

```diff
@@ -355,10 +374,19 @@ static __always_inline int __icmp6_handle_ns(struct __ctx_buff *ctx, int nh_off)
 	cilium_dbg(ctx, DBG_ICMP6_NS, target.p3, target.p4);
 
 	BPF_V6(router, ROUTER_IP);
-	if (ipv6_addrcmp(&target, &router) == 0) {
+	router_target = ipv6_addrcmp(&target, &router);
+
+	if (!router_target) {
```

Comment:
nit: Is there any reason to change these lines from the previous version?

`ipv6_addrcmp()` returns a u64, which is a bit unusual to convert into a bool. OTOH, the existing style of performing a comparison and checking against `0` is fairly standard compare logic in C.

Reply:
I might add here, this condition is actually the case where the target is the same as the router IP so checking `!router_target` implies that this is the case that *doesn't* handle the router target, but given the final bool passed to `send_icmp6_ndisc_adv()` below, we can infer that this really is the path that handles `to_router`. So probably worth clearing up this potential confusion for future readers.

Reply:
I think I copied from 
https://github.com/cilium/cilium/blob/8a03d54da11a1c15302f4f90de8bf1e539252784/bpf/lib/icmp6.h#L417

but I am you are right adding a `bool` here is not necessary. I will change it back to original shape.